includes/class-wc-payments-utils.php

```diff
@@ -619,4 +619,49 @@ public static function compose_transaction_url( $charge_id ) {
 			admin_url( 'admin.php' )
 		);
 	}
+
+	/**
+	 * Sanitize and retrieve a shortened statement descriptor concatenated with the order number.
+	 *
+	 * The shortened descriptor will be used as prefix and the order ID will be concatenated, acting as a suffix.
+	 *
+	 * @param string   $statement_descriptor Shortened statement descriptor.
+	 * @param WC_Order $order Order.
+	 * @return string $statement_descriptor Final shortened statement descriptor.
+	 */
+	public static function get_dynamic_statement_descriptor( $statement_descriptor = '', $order = null ) {
+		$statement_descriptor = self::clean_statement_descriptor( $statement_descriptor );
+
+		if ( method_exists( $order, 'get_order_number' ) && ! empty( $order->get_order_number() ) ) {
+			$statement_descriptor = $statement_descriptor . '* #' . $order->get_order_number();
+		}
+
+		// Limit it to 22 characters just in case.
+		$statement_descriptor = substr( $statement_descriptor, 0, 22 );
+
+		return $statement_descriptor;
+	}
+
+	/**
+	 * Sanitize statement descriptor text.
+	 *
+	 * Stripe requires max of 22 characters and no special characters.
+	 *
+	 * @since 4.0.0
```

Comment:
@cesarcosta99 I don't think we're officially applying `@since`/`@version` to the DocBlock in WCPay, besides leftovers from the Stripe extension (maybe?) 😅.

We even have a futuristic `@since 4.7.0` 🤖.

Reply:
🤖 I've seen many references of `@since` across the project, but I thought it was also a good call to remove them because those methods are very specific, and are not widely used. Removed them in 6d3422e and 86b05a7.